3610 Add ascontiguous utility and check in CacheDataset

Nic-Ma opened the pull request:
Fixes #3610 .

### Description
This PR added the `ascontiguous` utility and apply in `CacheDataset`.

### Status
**Ready**

### Types of changes
<!--- Put an `x` in all the boxes that apply, and remove the not applicable items -->
- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

Nic-Ma commented:
/black

Nic-Ma commented:
/build

Nic-Ma commented:
I verified the PR with spleen segmentation pipeline.

Thanks.

Nic-Ma commented on monai/transforms/utils.py:
```diff
@@ -1496,5 +1498,20 @@ def convert_pad_mode(dst: NdarrayOrTensor, mode: Union[NumpyPadMode, PytorchPadM
     raise ValueError(f"unsupported data type: {type(dst)}.")
 
 
+def as_contiguous(data):
```
I put this utility in the `transforms` folder because I think we may consider to add a `AsContiguous` transform later.

Thanks.

wyli reviewed (approved):
Thanks, it looks good to me. (Torch contiguous can take additional parameters but I'm not sure if we need it at the moment. https://pytorch.org/docs/stable/generated/torch.Tensor.contiguous.html#torch.Tensor.contiguous)

Nic-Ma commented:
> Thanks, it looks good to me. (Torch contiguous can take additional parameters but I'm not sure if we need it at the moment. https://pytorch.org/docs/stable/generated/torch.Tensor.contiguous.html#torch.Tensor.contiguous)

Thanks for your review, sure, I added `kwargs` for Tensor in the latest commit but didn't provide the arg for `CacheDataset` and explicitly set the `memory_format` in `CacheDataset`.

Thanks.

Nic-Ma commented:
/black

Nic-Ma commented:
/build

wyli commented:
Forgot to mention that 2 minor antipatterns were introduced: https://deepsource.io/gh/Project-MONAI/MONAI/run/a37b8f84-4ba2-4db2-b025-0ee5ee9ecc97/python/

Nic-Ma commented:
Hi @wyli ,

Thanks for your reminder, I updated to fix the antipatterns and changed the utility function name to align with:
https://github.com/Project-MONAI/MONAI/blob/dev/monai/utils/type_conversion.py#L102

Thanks.

Nic-Ma commented:
/black

Nic-Ma commented:
/build

Nic-Ma commented:
/build

wyli commented:
/build

wyli commented:
/build
